fastlane/lib/fastlane/actions/cocoapods.rb

```diff
@@ -100,13 +100,14 @@ def self.available_options
                                        description: 'A callback invoked with the command output if there is a non-zero exit status',
                                        optional: true,
                                        is_string: false,
+                                       type: :string_callback,
                                        default_value: nil),
           FastlaneCore::ConfigItem.new(key: :try_repo_update_on_error,
                                        env_name: "FL_COCOAPODS_TRY_REPO_UPDATE_ON_ERROR",
                                        description: 'Retry with --repo-update if action was finished with error',
                                        optional: true,
                                        is_string: false,
-                                       default_value: true,
```

Comment:
wasn't necessary?

Reply:
it was causing a conflict with `repo_update` since that is set to `false` as default.

Reply:
Shouldn't it be the other way around, i.e. only update the repo when there is an error?

Reply:
Yeah, but I don't think it's plumbed that way right now, so we'd have to update how the relatively newish `try_repo_update_on_error` interacts with the `repo_update` option.